mesa/datacollection.py

```diff
@@ -100,6 +101,8 @@ class attributes of model
 
         self.model_vars = {}
         self._agent_records = {}
+        self.agent_attr_index = {}
```

Comment:
This needs to be `self.agent_attr_indexes`, to make it consistent with existing dict attribute names, e.g. `self.model_vars`, etc.

Reply:
I thought of it as an index between reporter names and their position in the `_agent_records`. So a single index.

Maybe `agent_records_index` might be even clearer.

Reply:
In that case, the usage of the term "an index" as a collection of the mapping is inconsistent with its later use:
```
        # Get the index of the reporter
        attr_index = self.agent_attr_index[reporter]
```
Where this is a single index.

Reply:
Do you have a suggestion for a better name?

Reply:
@rht is the best at ensuring and understanding standards, so I am going to default to him. 

I tried a couple names to add something to this conversation but nothing captured it. I will keep thinking on it and update.